scopes/dependencies/dependency-resolver/dependency-resolver.docs.md

```diff
@@ -1,235 +1,322 @@
-    * (For historical reasons, JavaScript package managers generally do not treat this invalid
-    * state as an error.)
-    *
-    * The default value is false to avoid legacy compatibility issues.
-    * It is strongly recommended to set strictPeerDependencies=true.
-    */
-    "strictPeerDependencies": true,
-
-    /**
-     * map of extra arguments to pass to the configured package manager upon the installation
-     * of dependencies.
-    */
-    "packageManagerArgs": []
+For example, the following configuration will set `classnames` version `1.0.0` on all component using the `react-ui` namespace. This policy will override the workspace-level policy that uses version `2.0.0` of that same package.
+
+```json
+// All components using the namespace 'react-ui' will use version 1.0.0 of "classnames"
+// instead of version "2.0.0", set as the default for all components in the workspace
+{
+  "teambit.workspace/workspace": {
+    "name": "my-workspace"
+  },
+  "teambit.dependencies/dependency-resolver": {
+    "policy": {
+      "dependencies": {
+        "classnames": "2.0.0"
+      }
+    }
+  },
+  "teambit.workspace/variants": {
+    "teambit.dependencies/dependency-resolver": {
+      "policy": {
+        "{react-ui/*}": {
+          "classnames": "1.0.0"
+        }
+      }
+    }
+  }
+}
+```
+
+### "Forcibly" add dependencies to a component
+
+Dependency policies applied on a selected group of components will "forcibly" add the listed packages to any [selected] component that does not have them already listed as a dependency.
+This can be useful when a component depends on another module but has no `import`/`require` statement to be parsed by the Dependency Resolver (for example, in a Webpack configuration file).
+
+In the below example, `classnames@1.0.0` will be "forcibly" added as a dependency to any component using the `react-ui` namespace.
+
+```json
+  "teambit.workspace/variants": {
+    "teambit.dependencies/dependency-resolver": {
+      "policy": {
+        "{react-ui/*}": {
+          "classnames": "1.0.0"
+        }
+      }
+    }
   }
 ```
-### variant config
-
-The component configuration of the dependency resolver will store dependencies policy for the component.
-This is similar to what used to be under the `overrides.dependencies` in the old workspace config.
-The component configuration will support `"-"` sign as a value for dependency which means to remove it from the final dependency list.
-
-The component configuration will support syntax to remove all dependencies (by type) in case the user wants full control. (syntax is open for discussion)
-for example:
-```js
-"dependencies": {
-   // Remove all dependencies calculated from the workspace
-   "*": "-",
-   // Add custom dependency
-   "teambit.bit/my-awesome-component": "5.5.5"
-   },
+
+### Setting dependency types
+
+#### Setting dev dependencies
+
+Dev dependencies are determined by the type of file that consumes them.
```

Comment:
mention that if something is both required in dev file and regular file it will be considered as runtime not a dev